README.md

```diff
@@ -1,7 +1,334 @@
+* The call returns the generated ordering ID if you did not specify one. You can store it. It can be useful to delete a batch of documents.
+
+### Pushing a document with big properties
+In case the size of the document you just created is too big, the SDK does not handle it if you use the call to index a single document. If the whole document is larger than 256MB, it will throw an `ArgumentException`. If the document is larger than 5MB, it will throw a `CoveoPlatformException` including the status code (500) and `Internal server error` inside the `ErrorMessage` property. Take note that if you always use the batched call the SDK will handle it automatically.
+In that case you have 2 options:
+* You can add the document using the batched call `client.DocumentManager.AddOrUpdateDocuments`
+* You can also check for the document size and then decide if you use `client.DocumentManager.AddOrUpdateDocument` or `client.DocumentManager.AddOrUpdateDocuments`
+```csharp
+document.JsonObjectSize > Constants.COMPRESSED_DATA_MAX_SIZE_IN_BYTES
+```
+
+### Pushing a document with a large binary data
+The Push API has a hard limit of 5MB (compressed) for the document's binary data. When the data exceeds that value we need to request an upload URI to an S3 bucket to put the document's binary data in it and refer to that ID when pushing the document in the Push API. However, lucky you, the SDK handles this automatically. Thus, the SDK compresses the content, and if it exceeds 5MB (compressed), it will do the needed logic. Take note that the maximum size of a document is 256MB (compressed). 
+
+### Delete a single document
+```csharp
+string documentId = "https://coveo.com";
+client.DocumentManager.DeleteDocument(sourceId, documentId, null);
+```
+**Good to know:**
+* Prefer deleting documents in batches to save API calls and to have a better application performance.
+
+### Delete a batch of documents
+```csharp
+IList<string> documentsIdstoDelete = new List<string> {
+    "https://coveo.com",
+    "https://coveo.com/a page"
+};
+client.DocumentManager.DeleteDocuments(sourceId, documentsIdstoDelete, null);
+```
+
+### Delete a specific document and its children
+You can delete a specific document and all its children easily. For more information about deleting a document and its children, visit [Deleting an Item and Optionally, its Children in a Push Source](https://docs.coveo.com/en/171/cloud-v2-developers/deleting-an-item-and-optionally-its-children-in-a-push-source)
+
+In this example, imagine you have added a document with an ID of `http://coveo.com/parent` and another document with an ID of `http://coveo.com/parent/child`. Clearly, it this example, the second document is the child of the former. To delete a document and all its children, it is pretty easy:
```

Comment:
"To delete a document and all its children, it is pretty easy:" -> "To delete the parent and child documents:"

(it's generally a bad practice to mention that something is "easy" or "simple" in technical documentation. It tends to frustrate people who have issues, and does not really bring value to people who are not having issues)

Reply:
```suggestion
In this example, imagine you have added a document with an ID of `http://coveo.com/parent` and another document with an ID of `http://coveo.com/parent/child`. Clearly, it this example, the second document is the child of the former. To delete the parent and child documents:
```